gridstatus/eia.py

```diff
@@ -149,6 +154,222 @@ def fetch_page_wrapper(url, headers, page, page_size):
+                    break
+                try:
+                    rowspan = int(s1.get("rowspan"))
+                    if s1.select("a", class_="lbox"):
+                        rowspan -= 1  # down index by one (crack spread)
+                        s2 = s1.find_next_sibling("td", class_="s2").text
+                        d1 = s1.find_next_sibling("td", class_="d1").text
+                        direction = float(
+                            s1.find_next_sibling("td", class_=directions).text,
+                        )
+                        df_petrol.loc[len(df_petrol)] = (
+                            text,
+                            s2,
+                            d1,
+                            float(direction) if direction != "NA" else "NA",
+                        )
+                    else:
+                        for i in range(rowspan_sum, rowspan + rowspan_sum):
+                            s2_elements = parent.select("td.s2")
+                            d1_elements = parent.select("td.d1")
+                            direction_elements = parent.find_all(class_=directions)
+                            df_petrol.loc[len(df_petrol)] = (
+                                text,
+                                s2_elements[i].text,
+                                d1_elements[i].text,
+                                float(direction_elements[i].text),
+                            )
+
+                    rowspan_sum += rowspan
+                except TypeError:
+                    s2 = s1.find_next_sibling("td", class_="s2").text
+                    d1 = s1.find_next_sibling("td", class_="d1").text
+                    direction = float(
+                        s1.find_next_sibling("td", class_=directions).text,
+                    )
+                    df_petrol.loc[len(df_petrol)] = (
+                        text,
+                        s2,
+                        d1,
+                        float(direction) if direction != "NA" else "NA",
+                    )
+
+            natural_gas_spots = soup.select_one(
+                "table[summary='Spot Natural Gas and Electric Power Prices']",
+            )
+
+            for s1 in natural_gas_spots.select("td.s1"):
+                price_siblings = s1.find_next_siblings("td", class_="d1")
+                direction_siblings = s1.find_next_siblings("td", class_=directions)
+                df_ng.loc[len(df_ng)] = (
+                    s1.text,
+                    price_siblings[0].text,
+                    float(direction_siblings[0].text)
+                    if direction_siblings[0].text != "NA"
+                    else "NA",
+                    price_siblings[1].text,
+                    float(direction_siblings[1].text)
+                    if direction_siblings[1].text != "NA"
+                    else "NA",
+                    price_siblings[2].text,
+                )
+
+        df_ng["date"] = pd.to_datetime(close_date)
```

Comment:
is there a timezone that would make sense to add to this date?

Reply:
Potentially the central time zone, as the henry hub contract trades on CME (as seen here): https://www.cmegroup.com/markets/energy/natural-gas/natural-gas.calendar.html. Additionally, HH is geographically located in the Central time zone. And lastly, it's the most middle of the timezones.

Reply:
look at it more, it seems like a timezone doesn't make sense. i think all of these are the price at market close of that day. in theory, i tihnk that means it doesnt make sense to have the time portion. just a date with no timezone like you have for `delivery_month`, `week_ending_date` coal spot data

Reply:
sorry just saw this. will do